private-templates-service/client/src/store/search/types.ts

```diff
@@ -1,15 +1,30 @@
+import { TemplateList } from "adaptive-templating-service-typescript-node";
+import { IncomingMessage } from "http";
+
 export interface SearchState {
   isSearch: boolean;
   searchValue: string;
+  templates?: TemplateList; 
+  loading: boolean;
+  error?: IncomingMessage;
 }
 
 // Action Types 
-export const SEARCH = 'SEARCH';
+export const QUERY_SEARCH_BEGIN = 'QUERY_SEARCH BEGIN';
+export const QUERY_SEARCH_SUCCESS = "QUERY SEARCH SUCCESS";
+export const QUERY_SEARCH_FAILURE = "QUERY SEARCH FAILURE";
 export const CLEAR_SEARCH = 'CLEAR_SEARCH';
 
 // Actions
 export interface SearchAction{
-  type: typeof SEARCH | typeof CLEAR_SEARCH;
-  text: string
```

Comment:
What is the use for `text`?

Reply:
The text param is just for debugging the redux states. I will remove the text stuff